app/controllers/usages_controller.rb

```diff
@@ -0,0 +1,46 @@
+class UsagesController < ApplicationController
+  before_action :set_usage, only: [:edit, :update]
+  before_action :initialize_usage, only: [:new, :create]
+  before_action :set_subscription, only: [:new, :edit, :create, :index]
+  before_action :set_subscriptions_features, only: [:new, :edit]
+
+  def create
+    if @subscription.create_usage(usage_params)
+      flash[:success] = 'Usage Successfully Added'
+    else
+      flash[:danger] = 'Error occurred, Try Again'
+    end
+    redirect_to subscriptions_url
+  end
+
+  def update
+    if @usage.update(usage_params)
+      flash[:success] = 'Usage Successfully Updated'
+    else
+      flash[:danger] = 'Error occurred, Try Again'
+    end
+    redirect_to subscriptions_url
+  end
+
+  private
+
+  def usage_params
+    params.require(:usage).permit(:feature_id, :feature_count)
+  end
+
+  def initialize_usage
+    @usage = Usage.new
+  end
+
+  def set_usage
+    @usage = Usage.includes(:subscription, :feature).find_by_id(params[:id])
```

Comment:
there is no need to pre-load if you are only getting one object.
N+1 only becomes an issue when you try to get associated value of multiple objects.